chia/util/cached_bls.py

```diff
@@ -23,12 +23,18 @@ def get_pairings(cache: LRUCache, pks: List[bytes48], msgs: Sequence[bytes], for
             if missing_count > len(pks) // 2:
                 return []
         pairings.append(pairing)
-
+    pk_bytes_to_g1: Dict[bytes48, G1Element] = {}
     for i, pairing in enumerate(pairings):
         if pairing is None:
             aug_msg = pks[i] + msgs[i]
             aug_hash: G2Element = AugSchemeMPL.g2_from_message(aug_msg)
-            pairing = G1Element.from_bytes(pks[i]).pair(aug_hash)
+
+            if pks[i] in pk_bytes_to_g1:
```

Comment:
this seems like something that might happen often enough to change this to a single lookup, using `pk_bytes_to_g1.get()`

Reply:
Does the performance differ at all? Is this just a style comment?

Reply:
I would not assume that python can merge the two lookups into one. but maybe I'm just pessemistic.
The point is that if `get()` returns `not None` you can use that value rather than looking it up again.

Right now you do two lookups, first using `in` and then using `[]`.

Reply:
I also didn't realize at first this is just a cache for this one call, meaning it won't have more than 2000 elements in it, so maybe it won't ever matter.